khmer/_khmermodule.cc

```diff
@@ -1326,6 +1326,91 @@ static PyObject * hash_consume_and_tag(PyObject * self, PyObject * args)
     return Py_BuildValue("K", n_consumed);
 }
 
+static PyObject * hash_get_tags_and_positions(PyObject * self, PyObject * args)
+{
+    khmer_KCountingHashObject * me = (khmer_KCountingHashObject *) self;
+    CountingHash * counting = me->counting;
+
+    const char * seq;
+
+    if (!PyArg_ParseTuple(args, "s", &seq)) {
+        return NULL;
+    }
+
+    // call the C++ function, and trap signals => Python
+
+    std::vector<unsigned int> posns;
+    std::vector<HashIntoType> tags;
+    try {
+        unsigned int pos = 1;
+        KMerIterator kmers(seq, counting->ksize());
+
+        while (!kmers.done()) {
+            HashIntoType kmer = kmers.next();
+            if (set_contains(counting->all_tags, kmer)) {
+                 posns.push_back(pos);
+                 tags.push_back(kmer);
+            }
+            pos++;
+        }
+    } catch (_khmer_signal &e) {
+        PyErr_SetString(PyExc_ValueError, e.get_message().c_str());
+        return NULL;
+    }
+
+    PyObject * posns_list = PyList_New(posns.size());
+    for (size_t i = 0; i < posns.size(); i++) {
+        PyObject * tup = Py_BuildValue("IK", posns[i], tags[i]);
+        PyList_SET_ITEM(posns_list, i, tup);
+    }
+
+    return posns_list;
+}
+
+static PyObject * hash_find_all_tags_list(PyObject * self, PyObject *args)
+{
+    khmer_KCountingHashObject * me = (khmer_KCountingHashObject *) self;
+    CountingHash * counting = me->counting;
+
+    const char * kmer_s = NULL;
+
+    if (!PyArg_ParseTuple(args, "s", &kmer_s)) {
+        return NULL;
+    }
+
+    if (strlen(kmer_s) < counting->ksize()) {
+        PyErr_SetString( PyExc_ValueError,
+                         "starting kmer is smaller than the K size of the counting table");
+        return NULL;
+    }
+
+    SeenSet tags;
+
+    Py_BEGIN_ALLOW_THREADS
+
+    HashIntoType kmer, kmer_f, kmer_r;
+    kmer = _hash(kmer_s, counting->ksize(), kmer_f, kmer_r);
```

Comment:
GCC [complained][1] about kmer being defined but not used in this function, do you see it being used in the future? If not maybe just ignore the return value and remove it from the previous line?

[1]: http://ci.ged.msu.edu/job/khmer-pullrequest/1134/label=linux/warnings23Result/new/